package.json

```diff
@@ -2,15 +2,7 @@
   "name": "visible-wordpress",
   "version": "1.0.0",
   "scripts": {
-    "export-db": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp db export /data/database.sql\"",
-    "import-db": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp db import /data/database.sql\" && npm run search-replace",
-    "update-plugins": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp plugin update --all\"",
-    "update-themes": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp theme update --all\"",
-    "plugin-install": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp plugin install $PLUGIN\"",
-    "plugin-update": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp plugin update $PLUGIN\"",
-    "theme-update": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp theme update $THEME\"",
-    "theme-install": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp theme install $THEME\"",
-    "search-replace": "docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp search-replace 'http://www.oldurl.com' 'http://localhost:8080' --skip-columns=guid\""
+    "wp": "wpcmd() { docker exec $(docker ps -lq) /bin/bash -c \"sudo -u www-data wp $(echo $@)\"; };wpcmd "
```

Comment:
Hi!

Very nice solution having one command to use the wp-cli! Very helpfull in the development workflow! I do have some problems with this sollution though. Number one: it does not work on Windows.

There are two problems regarding to windows:
- You can't define functions like that on the cmd prompt
- You can't substitute a command's output like with $(docker ps -lq).

The only possible solution I can think of right now is to make a script or separate npm module that does this calling.

Second problem I found with this method is that you can't be sure that `docker ps -lq` outputs the right last made container. For example, if you run the `docker-compose up` command, play around a bit. And after that run another docker command like `docker run hello-world`, the docker ps -lq outputs the last container made: the hello-world container.

You can also run `docker-compose ps` from the folder with the docker-compose.yml file. This command seems to output only the containers that are defined from the yml file. However you still need a way to verify the right container to send the command to. You could do some smart guessing, docker-compose takes the folder name of the project + the service name + a sequence number as the container name. Maybe if you could point to the WordPress container only and send the command there, your done. Even if you would have more WordPress containers (in a scaled environment) this would most probably not be a problem because the database would be separate (on the other hand this would only be the case in production. Are you going to use wp-cli there? Preferably not!).

You could use the container_name directive in the .yml file. This way you would be certain about the container its name and use that safely. This is considered a bad practice: docker requires containers to have unique names and this would get in your way when you would need to scale.

That made me think. Maybe it's perfectly legit to define a separate service in the .yml file. a named container just for the wp-cli tool. For development that would be perfectly fine, just don't push it to production.

I really like this being part of the wordpress-starter project. I do feel the need to say that it must be more a more stable solution than it is now. Otherwise it would be better to document the possibility to check the container name for yourself (as a developer) and then run the docker exec command for yourself.